### ISSUE TITLE
API for summary of changes

### ISSUE THREAD

**Issue by gwarnes-mdsol** [paulfitz/daff#87]
Hi, 

TLDR:  Please add an API for summarizing changes

Details:

I've added code to the R interface (edwindj/daff via pulls from gwarnes-mdsol/daff) to generate a summary table of the number of added/removed/changed rows and columns in a `TableView` filled by `TableDiff.highlite`, e.g:

```
Data diff:
 Comparison: ‘y’ vs. ‘x’ 
        #       Changed Removed Added
Rows    3 --> 2 1       1       0    
Columns 5       0       1       1  
```

This is currently accomplished by identifying the row and column containing flags in Table.get_data() and then counting the occurrences of each flag type.  This is inefficient and error-prone.

Please add appropriate API calls to obtain this information.  Perhaps one or more methods to `TableDiff`?  Something like:

```
TableDiff.getNumAddedRows(): Int
TableDiff.getNumAddedColumns(): Int

TableDiff.getNumDeletedRows(): Int
TableDiff.getNumDeletedColumns(): Int

TableDiff.getNumModifiedRows(): Int
TableDiff.getNumModifiedColumns(): Int

TableDiff.getNumReorderedRows(): Int
TableDiff.getNumReorderedColumns(): Int

TableDiff.getNumRows(): Array<Int>      /* length two:  ( local, remote ) */
TableDiff.getNumColumns(): Array<Int> /* length two:  ( local, remote ) */
```

**Comment by paulfitz** [comment]
Makes sense, I'll see what I can do.  Thanks for the suggestion!

**Comment by gwarnes-mdsol** [comment]
Works well for me.
